src/Text.tsx

```diff
@@ -1,13 +1,53 @@
+import React from 'react'
 import styled from 'styled-components'
 import {COMMON, SystemCommonProps, SystemTypographyProps, TYPOGRAPHY} from './constants'
+import {ForwardRefComponent as PolymorphicForwardRefComponent} from './utils/polymorphic'
 import sx, {SxProp} from './sx'
-import {ComponentProps} from './utils/types'
 
-const Text = styled.span<SystemTypographyProps & SystemCommonProps & SxProp>`
+const restrictedAs = [
+  'span',
+  'p',
+  'b',
+  'i',
+  'em',
+  'small',
+  'strong',
+  'u',
+  'code',
+  'q',
+  's',
+  'label',
+  'ins',
+  'del',
+  'mark',
+  'kbd',
+  'abbr',
+] as const
+
+const StyledText = styled.span<SystemTypographyProps & SystemCommonProps & SxProp>`
   ${TYPOGRAPHY};
   ${COMMON};
   ${sx};
 `
 
-export type TextProps = ComponentProps<typeof Text>
+export type TextProps = {
+  as?: typeof restrictedAs[number]
```

Comment:
Does `keyof restrictedAs` work here?

Reply:
I'm getting an error here when using `keyof`, but I think it's because I changed `restrictedAs` from an object to an array.